contracts/eosio.bios/include/eosio.bios/eosio.bios.hpp

```diff
@@ -254,6 +294,23 @@ namespace eosiobios
 
       typedef eosio::multi_index<"abihash"_n, abi_hash> abi_hash_table;
 
+      struct [[eosio::table]] resource_config {
+         uint64_t key; // This will be the primary key
+         uint64_t ram_price;
+         uint64_t total_ram_available;
+         uint64_t total_ram_used;
+         uint64_t total_cpu_weight_allocated;
+         uint64_t total_net_weight_allocated;
+         
+         uint64_t primary_key() const { return key; } 
+
+         EOSLIB_SERIALIZE(resource_config, (ram_price)(total_ram_available)(total_ram_used)
+         (total_cpu_weight_allocated)(total_net_weight_allocated)
+         )
+      };
+
+      typedef eosio::multi_index<"resconfig"_n, resource_config> resource_config_table;
```

Comment:
suggested to use a singleton, instead of a table. This has a simpler API for a single-line table
https://docs.eosnetwork.com/docs/latest/smart-contracts/state-data#singleton-table
https://docs.antelope.io/cdt/latest/how-to-guides/multi-index/how-to-define-a-singleton

Reply:
e.g. singletons dont need a primary key